charts/prometheus-fastly-exporter/templates/deployment.yaml

```diff
@@ -0,0 +1,85 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "prometheus-fastly-exporter.fullname" . }}
+  labels:
+    app: {{ template "prometheus-fastly-exporter.name" . }}
+    chart: {{ template "prometheus-fastly-exporter.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app: {{ template "prometheus-fastly-exporter.name" . }}
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "prometheus-fastly-exporter.name" . }}
+        release: {{ .Release.Name }}
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+    spec:
+      serviceAccountName: {{ template "prometheus-fastly-exporter.serviceAccountName" . }}
+      {{- with .Values.initContainers }}
+      initContainers: {{ toYaml . | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args:
+            {{- if not .Values.envTokenFromSecret }}
+            - "-token={{ .Values.token }}"
+            {{- end }}
+            {{- range $key, $value := .Values.options }}
```

Comment:
why is this in a loop?
can there be multiple tokens?

can we do it similar to: https://github.com/prometheus-community/helm-charts/blob/5ce650b75c3bb4cc85e0f7c23ac2614bb37d67de/charts/prometheus-mongodb-exporter/templates/deployment.yaml#L31-L35

So the token is always stored and referenced from a secret and also the key in the secret is configurable?

More common naming for the variable would be "existingSecret".

Reply:
Thanks for the ref. 

The token is not in a loop.
The loop contains keys that can be passed like this: https://github.com/fastly/fastly-exporter#filtering-services

Reply:
And no, there must be only one token.

I slightly modified the definition of the variable.
Now it is always set from env: `FASTLY_API_TOKEN`